agents-api/agents_api/dependencies/developer_id.py

```diff
@@ -32,9 +32,9 @@ async def get_developer_id(
 async def get_developer_data(
     x_developer_id: Annotated[UUID | None, Header(include_in_schema=False)] = None,
 ) -> Developer:
-    if not multi_tenant_mode:
-        assert not x_developer_id, "X-Developer-Id header not allowed in multi-tenant mode"
-        return await get_developer(developer_id=UUID("00000000-0000-0000-0000-000000000000"))
+    # if not multi_tenant_mode:
```

Comment:
I don't think this is correct. `get_developer_data` is used by [session/chat](https://github.com/julep-ai/julep/blob/3eb38579e6483d48e1b77d482904922c49090adb/agents-api/agents_api/routers/sessions/chat.py) and [session/render](https://github.com/julep-ai/julep/blob/3eb38579e6483d48e1b77d482904922c49090adb/agents-api/agents_api/routers/sessions/render.py#L32) routes. And in single tenant mode, if this part is commented out, those routes will fail.

Reply:
@Ahmad-mtos this is the correct behavior but, if tests are failing, make sure that tests are not passing `X-Developer-Id` header when in _single-tenant_ mode.